controlplane/pkg/metrics/metrics_collector.go

```diff
@@ -0,0 +1,95 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc"
+	rpc "github.com/ligato/vpp-agent/api/configurator"
+	interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/crossconnect"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"io"
+	"time"
+)
+
+type MetricsCollector struct {
+	requestPeriod time.Duration
+}
+
+func NewMetricsCollector() *MetricsCollector {
+	return &MetricsCollector{
+		requestPeriod: time.Second * 15,
```

Comment:
Should we parametrise this with ENV variable?

Reply:
Good idea, but can be done in a followup PR :)